### ISSUE TITLE
Default the database on AppEngine server not require a password?

### ISSUE THREAD

**Issue by redhat-raptor** [GoogleCloudPlatform/appengine-django-skeleton#26]
Looking at the following code, `'PASSWORD': '<your-database-password>'` is only assigned to local settings. Should the PASSWORD field not be added for the AppEngine settings as well? 

> if os.getenv('SERVER_SOFTWARE', '').startswith('Google App Engine'):
>     # Running on production App Engine, so use a Google Cloud SQL database.
>     DATABASES = {
>         'default': {
>             'ENGINE': 'django.db.backends.mysql',
>             'HOST': '/cloudsql/<your-project-id>:<your-cloud-sql-instance>',
>             'NAME': '<your-database-name>',
>             'USER': 'root',
>         }
>     }
> else:
>     DATABASES = {
>         'default': {
>             'ENGINE': 'django.db.backends.mysql',
>             'NAME': '<your-database-name>',
>             'USER': '<your-database-user>',
>             'PASSWORD': '<your-database-password>',
>             'HOST': '<your-database-host>',
>             'PORT': '3306',
>         }
>     }
>

**Comment by nitrax92** [comment]
You bring up a good point, but it is not needed. 

When the  `if os.getenv('SERVER_SOFTWARE', '').startswith('Google App Engine'):` is true, the Application is run from Googles servers. When setting up the database, you can choose which application is allowed to access that Cloud SQL instance based on the application ID. So when the application is authorized from the database instance, that serves as enough authentication to allow the connection. (This is done under the Access Control tab within the Cloud SQL instance.)

**Comment by waprin** [comment]
What @nitrax92 said. App Engine talks to CloudSQL through a Unix root socket that only it can use, so no auth is needed. Maybe if you were doing something fancy where App Engine can only talk to certain tables/databases on the instance, but I didn't want to overcomplicate things.

If you have any issues actually connecting, file a new issue, but this is works-as-intended.

**Comment by redhat-raptor** [comment]
Thanks for the explanation! I actually had to set the password for the AppEngine env in order to access the SQL instance. Perhaps I did not configure the Access Control correctly.

But one thing sounds magic to me: Both TCP and Unix socket are ways to connect to the SQL instance... However when I reach the instance using unix socket and GAE firewall allows my app to access the instance... I thought SQL would still ask for the password! At least traditional MySQL instances have this behaviour as of my understanding! 

:+1:
